src/pudl/analysis/record_linkage/name_cleaner.py

```diff
@@ -56,69 +70,109 @@ class LegalTermLocation(enum.Enum):
     ANYWHERE = 2
 
 
-class CompanyNameCleaner(BaseModel):
-    """Class to normalize/clean up text based company names."""
-
-    # Constants used internally by the class
-    __NAME_LEGAL_TERMS_DICT_FILE = "us_legal_forms.json"
-    __NAME_JSON_ENTRY_LEGAL_TERMS = "legal_forms"
-
-    #: A flag to indicate if the cleaning process must normalize
-    #: text's legal terms. e.g. LTD => LIMITED.
-    cleaning_rules_list: list[str] = [
-        "remove_word_the_from_the_end",
-        "remove_word_the_from_the_beginning",
-        "replace_amperstand_between_space_by_AND",
-        "replace_hyphen_by_space",
-        "replace_hyphen_between_spaces_by_single_space",
-        "replace_underscore_by_space",
-        "replace_underscore_between_spaces_by_single_space",
-        "remove_all_punctuation",
-        "remove_numbers",
-        "remove_math_symbols",
-        "remove_words_in_parentheses",
-        "remove_parentheses",
-        "remove_brackets",
-        "remove_curly_brackets",
-        "enforce_single_space_between_words",
-    ]
-
-    #: A flag to indicate if the cleaning process must normalize
-    normalize_legal_terms: bool = True
-
-    #: Define if unicode characters should be removed from text's name
-    #: This cleaning rule is treated separated from the regex rules because it depends on the
-    #: language of the text's name. For instance, russian or japanese text's may contain
-    #: unicode characters, while portuguese and french companies may not.
-    remove_unicode: bool = False
+class Lettercase(enum.Enum):
+    """Allowed cases for output strings."""
 
-    #: Define the letter case of the cleaning output
-    output_lettercase: Literal["lower", "title"] = "lower"
+    LOWER = 1
+    TITLE = 2
+    UPPER = 3
 
-    #: Where in the string are legal terms found
-    legal_term_location: LegalTermLocation = LegalTermLocation.AT_THE_END
 
-    #: Define if the letters with accents are replaced with non-accented ones
+class HandleLegalTerms(enum.Enum):
+    """Whether to leave, remove, or normalize legal terms."""
+
+    NORMALIZE = 3
+    LEAVE_AS_IS = 1
+    REMOVE = 2
+
+
+def _get_legal_terms_dict() -> dict[str, list]:
+    json_source = files("pudl.package_data.settings").joinpath(
+        NAME_LEGAL_TERMS_DICT_FILE
+    )
+    with json_source.open() as json_file:
+        legal_terms_dict = json.load(json_file)[NAME_JSON_ENTRY_LEGAL_TERMS]["en"]
+    return legal_terms_dict
+
+
+class CompanyNameCleaner(BaseModel):
+    """Class to normalize/clean up text based company names.
+
+    Attributes:
```

Comment:
I resolve the warning/error you need to move the attribute doc strings directly below the attribute definitions:

```python
cleaning_rules_list: list[str] = DEFAULT_CLEANING_RULES_LIST
    """A list of cleaning rules that the
    
    CompanyNameCleaner should apply. Will be validated to ensure
    rules comply to allowed cleaning functions."""
```

You can use this [pudl class](https://github.com/catalyst-cooperative/pudl/blob/c33a675d9931ef96ab8c58461ab920bba0dec5f9/src/pudl/metadata/classes.py#L328) as a reference. 

I think you're getting a duplicate warning because [sphinx is creating empty doc strings](https://github.com/sphinx-doc/sphinx/issues/8664#issuecomment-757087741) for the attributes and then it looks at the doc string of the class and creates doc strings for the attributes again.